app/src/services/datasetService.js

```diff
@@ -63,33 +64,43 @@ class DatasetService {
+                                alertEndDate: new Date().toISOString().slice(0, 10),
+                                alertResult: result.data[0].value
+                            };
+
+                            // Execute EMAIL notification - sending an email
                             if (subscription.resource.type === 'EMAIL') {
-                                const areaName = subscription.params.area ? await DatasetService.getAreaName(subscription.params.area) : '';
-                                const datasetName = metadata[0].attributes.info.name || dataset.attributes.name || dataset.name;
-                                const data = {
-                                    subject: subscription.name ? subscription.name : `${datasetQuery.type} in ${areaName} above ${datasetQuery.threshold}`,
-                                    datasetName,
-                                    datasetId: datasetQuery.id,
-                                    datasetSummary: metadata[0].attributes.info.functions,
-                                    areaId: subscription.params.area ? subscription.params.area : '',
-                                    areaName,
-                                    alertName: subscription.name ? subscription.name : `${datasetQuery.type} in ${areaName} above ${datasetQuery.threshold}`,
-                                    alertType: datasetQuery.type,
-                                    alertBeginDate: datasetQuery.lastSentDate.toISOString().slice(0, 10),
-                                    alertEndDate: new Date().toISOString().slice(0, 10),
-                                    alertResult: result.data[0].value
-                                };
                                 logger.debug('Sending mail with data', data);
                                 let template = 'dataset-rw';
                                 if (subscription.env && subscription.env !== 'production') {
                                     template += `-${subscription.env}`;
                                 }
                                 MailService.sendMail(template, data, [{ address: subscription.resource.content }], 'rw'); // sender='rw'
+                            }
 
-                            } else {
-                                // @TODO resource.type === 'WEBHOOK'?
+                            // Execute URL notification - POSTing to webhook
+                            if (subscription.resource.type === 'URL') {
+                                // POST to URL configured in subscription.resource.content
+                                request.post(subscription.resource.content, { json: data }, (error, res, body) => {
+                                    logger.debug('Response received from POST to subscription web-hook: ', res.statusCode, body);
+                                    if (error) {
+                                        logger.error('Error POSTing to subscription web-hook: ', error);
```

Comment:
logger.error should be used for when the code fails. Let's try to parse what http code we get back on error, and log accordingly. For example, a 4xx is not an app error, and thus should not be a logger.error.

Reply:
Improved error handling - now parsing the status code, logging just a `debug` message in case of a non-200 status code. Other cases (such as an invalid URL) still log an `error` message, since that is indeed an error

Reply:
keep in mind that these may be called in multiple succession by multiple subscriptions. So, for example, let's say a single user is reporting that their webhooks are not being called, and you check to logs to see what's happening. Can you identify that user's log lines from the info you are outputting, and tell it apart from all the other users lines?

Reply:
That's right :+1: 
Added the `subscription` object to the log message, I believe it contains all the data necessary for identifying who made the request.